src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -15918,8 +15933,11 @@ static void Report<T>(MyCollection<T> c)
                     }
                 }
                 """;
-            comp = CreateCompilation(new[] { sourceB, s_collectionExtensions }, references: new[] { refA }, targetFramework: TargetFramework.Net60, options: TestOptions.ReleaseExe);
+            comp = CreateCompilation(new[] { sourceB, s_collectionExtensions }, references: new[] { refA }, targetFramework: TargetFramework.Net60, options: TestOptions.ReleaseExe.WithSpecificDiagnosticOptions(WithSpanAllocWarning));
             comp.VerifyEmitDiagnostics(
+                // 0.cs(12,60): warning CS9208: Collection expression of type 'MyCollection<T>' may incur unexpected heap allocations. Consider explicitly creating an array, then converting to 'MyCollection<T>' to make the allocation explicit.
```

Comment:
Since the user-defined ref structs are always exposing some Create(ROS<...>) method, we know the ref struct can always be created from an array. You're right that `MyCollectionBuilder.Create((T[])[x, y, z])` can be used in this case. I could include the symbol in the diagnostic message for reference if you think that would be helpful.